src/c2goto/library/pthread_lib.c

```diff
@@ -275,6 +278,23 @@ __ESBMC_HIDE:;
   return 0;
 }
 
+// The pthread_detach() marks the thread identified by thread as detached.
+// source: https://man7.org/linux/man-pages/man3/pthread_detach.3.html
+int pthread_detach(pthread_t threadid)
+{
+__ESBMC_HIDE:;
+  __ESBMC_atomic_begin();
+
+  _Bool det = __ESBMC_pthread_thread_detach[(int)threadid];
+  // abort execution when thread is detached;
+  __ESBMC_assert(
+    det != 0, "You want to detach a thread which already detached");
+  __ESBMC_terminate_thread();
+  __ESBMC_pthread_thread_detach[(int)threadid] = 1;
+  __ESBMC_atomic_end();
+
+  return 0;
```

Comment:
The reference defines three return values:

- 0: When no error occurred
- EINVAL:  thread is not a joinable thread.
- ESRCH:  No thread with the ID thread could be found.